src/guide/essentials/template-refs.md

```diff
@@ -1,25 +1,26 @@
 # Template Refs
 
-While Vue's declarative rendering model abstracts away most of the direct DOM operations for you, there may still be cases where we need direct access to the underlying DOM elements. To achieve this, we can use the special `ref` attribute:
+Vue ning deklarativ renderlash modeli(template sintaksisi( `{}` ) orqali ma'lumotlarni DOMga chiqarish) siz uchun DOM ning to'g'ridan-to'g'ri bajariladigan ko'p operatsiyalarini(Masalan, `document.querySelector()`) qisqartirib bergan bo'lsa-da, DOM 
+
 
 ```vue-html
 <input ref="input">
 ```
 
-`ref` is a special attribute, similar to the `key` attribute discussed in the `v-for` chapter. It allows us to obtain a direct reference to a specific DOM element or child component instance after it's mounted. This may be useful when you want to, for example, programmatically focus an input on component mount, or initialize a 3rd party library on an element.
+`ref` - `key` atributiga o'xshagan maxsus atribut bo'lib, bu `v-for` bo'limida muhokama qilingan. Bu bizga DOM ning xos elementi yoki bola komponent namunasi(instance) `mounted` bo'lgandan keyin, to'g'ridan-to'g'ri aloqa yaratish imkonini beradi. Misol uchun, siz component `mount` bo'lganda `input` bilan bog'lanishni xohlasangiz yoki elementda yordamchi kutubxonani ishga tushirganingizda foydali bo'lishi mumkin.
 
-## Accessing the Refs
+## Ref dan foydalanish
 
 <div class="composition-api">
 
-To obtain the reference with Composition API, we need to declare a ref with the same name:
-
+Composition API bilan aloqa yaratish uchun, biz `ref` xuddi shu nom bilan e'lon qilishimiz kerak:
```

Comment:
![image](https://user-images.githubusercontent.com/12369295/172146689-be015f89-90b3-47c4-beb5-d44d7f2e4f0b.png)
`ref` dan keyi ni qo'shimchasi bo'lsa tushunarliroq bo'lardi. O'zbek tili grammatikasi bo'yicha

Reply:
To'g'rilandi.